src/Canvas/Canvas.js

```diff
@@ -0,0 +1,213 @@
+import * as React from "react";
+
+import { Box } from "@mui/material";
+import Grid from "./Grid";
+import PropTypes from "prop-types";
+import ResizeHandle from "./ResizeHandle";
+import SelectionRectangle from "./SelectionRectangle";
+import useSelectionRectangle from "./useSelectionRectangle";
+
+/**
+ * A canvas component that can be used for as part of a drawing application. It includes features such as resize, grid, and selection rectangle.
+ */
+const Canvas = React.forwardRef(
+  (
+    {
+      backgroundColor = "white",
+      backgroundImage = "",
+      border = true,
+      children,
+      grid = true,
+      gridColor = "rgba(0, 0, 0, 0.1)",
+      gridSize = 25,
+      height = 500,
+      minHeight = 100,
+      minWidth = 100,
+      onResize,
+      onSelectionRectangle,
+      resizable = true,
+      width = 500,
+      sx = [],
+      onMouseDown,
+      ...boxProps
+    },
+    ref
+  ) => {
+    // selection rectangle
+    const [isSelecting, rectangle, startSelection] = useSelectionRectangle(
+      onSelectionRectangle,
+      {
+        maxHeight: Math.max(height, minHeight),
+        maxWidth: Math.max(width, minWidth)
+      }
+    );
+
+    const handleMouseDown = e => {
+      onMouseDown && onMouseDown(e);
+      startSelection(e);
+    };
+
+    const handleResize = (width, height) => {
+      width = Math.max(width, minWidth);
+      height = Math.max(height, minHeight);
+      onResize(width, height);
+    };
+
+    return (
+      <Box
+        sx={[
+          {
+            backgroundColor,
+            backgroundImage: `url(${backgroundImage})`,
+            backgroundRepeat: "no-repeat",
+            backgroundSize: `${width}px ${height}px`,
+            border: border ? `1px solid ${gridColor}` : "none",
+            height: `${height}px`,
+            minHeight,
+            minWidth,
+            outline: "none",
+            position: "relative",
+            width: `${width}px`
+          },
+          ...(Array.isArray(sx) ? sx : [sx]) // combine user provided sx with default
```

Comment:
Is this the best way to do this? I prefer the MUI pattern where their compound components offer specific props for styling specific underlying base components. E.g. TextField accepts InputProps, SelectProps etc. Also, what are you anticipating needing to be passed in here via sx?

Reply:
We discussed that this is one of multiple solutions to custom styling available. We should align as a team and roll a solution out across the components.

An example usecase in this component is the desire to set overflow:hidden in some specific cases. We agree that providing customisation either via sx, custom theming, or sub component props would be useful. It is hard to anticipate use cases and this provides flexibility in using components going forward.

Reply:
Raised #462

Reply:
On the same topic here I have just noticed you are also allowing any random other props to be passed in and forwarded on to the box. Again, what's the reasoning for this and how would it be used? It feels like we're just adding something that could get abused and end up biting us in the future. Just add props that we know are needed today.